### PR TITLE
ansible: add linux-only dependencies

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
For the two ansible formulas, how will that work when poetry is used for updates?

**Comment by iMichka** [comment]
There are a bunch of other formulae with Mac/Linux only python dependencies. I have no clue of poetry. What is the impact?

**Comment by fxcoudert** [comment]
These formulas are generated automatically by poetry (as I understand it). So I don't think it makes sense to make manual modifications if they are going to be erased at the next update.

**Comment by SMillerDev** [comment]
I don't think the ansible formulae specifically can use poetry because of the optional dependencies we use. But either way this should be supported in brew as well.

CC @Rylan12 who made the initial version of the python dependency magic

**Comment by Rylan12** [comment]
Currently, running `brew bump-formula-pr` will automatically call the `brew update-python-resources` command. That command uses `pipgrip` to automatically determine which PyPI packages are needed as resources and modified/adds resource blocks as needed. This only runs on formulae that use a `files.pythonhosted.org` URL and are not on the [blocklist](https://github.com/SeekingMeaning/brew/blob/4cc111ce6d1547410107f175a98ca0b4bec4a6e6/Library/Homebrew/utils/pypi.rb#L12-L24). Conveniently, the `ansible` formulae don't fall into the auto-update category so this shouldn't affect them.

In general, will the resource blocks be different on macOS vs on Linux? If so, it's probably worth modifying that code to account for that. My guess, though, would be that for most formulae it would not be an issue. Even if some do my guess is that `update-python-resources` will fail when trying to update resources and will prompt the user to do so manually. Might be worth doing some experimentation, though.

**TL;DR**: No, this won't be an issue for the `ansible` formulae because they already special cases.

**Comment by iMichka** [comment]
I was able to fix the ansible issue, actually it was just krb5 missing on Linux. So I do not need to touch the resources, all is good.

For `ansible@2.8` it looks I need to add `importlib-metadata<3,>=0.12`

For `a2ps`, it fails to build and has not gotten any updates since 2007.
```
==> Analytics
install: 79 (30 days), 220 (90 days), 1,064 (365 days)
install-on-request: 79 (30 days), 220 (90 days), 1,061 (365 days)
build-error: 0 (30 days)
```

Should I rather deprecate it?

**Comment by SMillerDev** [comment]
deprecation seems fine to me

**Comment by iMichka** [comment]
I disabled `a2ps` in #64475.

**Comment by iMichka** [comment]
The ansible build error was fixed by #64831.
